hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/HddsDispatcher.java

```diff
@@ -619,7 +617,7 @@ private boolean isVolumeFull(Container container) {
     if (isOpen) {
       HddsVolume volume = container.getContainerData().getVolume();
       SpaceUsageSource usage = volume.getCurrentUsage();
-      long volumeFreeSpaceToSpare = freeSpaceCalculator.get(usage.getCapacity());
+      long volumeFreeSpaceToSpare = volume.getFreeSpaceToSpare(usage.getCapacity());
```

Comment:
It's fine, just not looking that straightforward that need a capacity parameter as the volume has the capacity value.

Reply:
Yes, I also don't like to pass capacity to volume, so I plan to further refactor in follow-up.  But wanted to avoid bloating this patch.

Reply:
Created HDDS-12670 for follow-up.

Reply:
@ChenSammi @peterxcli #8167 addresses this.